src/benchmarks/gc/src/exec/GCPerfSim/GCPerfSim.cs

```diff
@@ -2193,129 +2271,54 @@ private void DoSurvive(ITypeWithPayload item, in ObjectSpec spec)
                 oldArr.Replace(rand.GetRand(oldArr.Length), item);
                 break;
             case ItemType.ReferenceItem:
-                // For ref items we want to create some variation, ie, we want
-                // to create different graphs. But we also want to keep our live 
-                // data size the same. So we do a few different operations -
-                // If the live data size is the same as what we set, we randomly
-                // choose an action which can be one of the following -
-                // 
-                // 1) create a new item, take a few items off the array and link them onto 
-                // the new item.
-                // 
-                // 2) create a new item and a few extra ones and link them onto the new item.
-                // note this may not have much affect in ephemeral GCs 'cause it's very likely 
-                // they all get promoted to gen2.
-                // 
-                // 3) replace a bunch of entries with newly created items.
-                // 
-                // If the live data size is > what's set, we randomly choose a non null entry 
-                // and set it to null.
-
-                if (oldArr.TotalLiveBytes < curPhase.totalLiveBytes)
-                {
-                    MixItUp((ReferenceItemWithSize)item, spec);
-                }
-                else
+                // Step 1: Decide which list to pick a victim to free
```

Comment:
>               // Step 1: Decide which list to pick a victim to fr [](start = 1, length = 66)

this changes what it used to do.. let's talk about it in our meeting.

Reply:
could you please open an issue to add back the variability?

---
In reply to: [584232243](https://github.com/dotnet/performance/pull/1582#discussion_r584232243) [](ancestors = 584232243)

Reply:
https://github.com/dotnet/performance/issues/1712